daemon/mgr/image.go

```diff
@@ -161,12 +169,80 @@ func (mgr *ImageManager) GetImage(ctx context.Context, idOrRef string) (*types.I
 }
 
 // ListImages lists images stored by containerd.
-func (mgr *ImageManager) ListImages(ctx context.Context, filter ...string) ([]types.ImageInfo, error) {
-	// TODO: support filter functionality
+func (mgr *ImageManager) ListImages(ctx context.Context, filter filters.Args) ([]types.ImageInfo, error) {
 	ctrdImageInfos := mgr.localStore.ListCtrdImageInfo()
 	imgInfos := make([]types.ImageInfo, 0, len(ctrdImageInfos))
 
+	if err := filter.Validate(acceptedImageFilterTags); err != nil {
+		return nil, err
+	}
+
+	var (
+		beforeFilter, sinceFilter *types.ImageInfo
+		beforeTime, sinceTime     time.Time
+		err                       error
+	)
+
+	beforeImages := filter.Get("before")
+	if len(beforeImages) > 0 {
+		// use the last one if multi before images was assigned
+		beforeFilter, err = mgr.GetImage(ctx, beforeImages[len(beforeImages)-1])
```

Comment:
so, could we limit the multi-since/multi-before here? Since there is no reason for this requirement.

Reply:
@fuweid I test the same scenario in moby,  have the same problem.  I think the single-since/since-before is reasonable, while multi is undefined.  So in here, I want to throw err if len > 1, wdyt?

Reply:
agree